rust/drivers/datafusion/src/lib.rs

```diff
@@ -768,13 +892,31 @@ impl Statement for DataFusionStatement {
     }
 
     fn execute_update(&mut self) -> adbc_core::error::Result<Option<i64>> {
-        self.runtime.block_on(async {
-            let _ = self
-                .ctx
-                .sql(&self.sql_query.clone().unwrap())
-                .await
-                .unwrap();
-        });
+        if self.sql_query.is_some() {
+            self.runtime.block_on(async {
+                let _ = self
+                    .ctx
+                    .sql(&self.sql_query.clone().unwrap())
+                    .await
+                    .unwrap();
+            });
+        } else if self.bound_record_batch.get_mut().is_some() {
+            self.runtime.block_on(async {
+                let batch: RecordBatch = self.bound_record_batch.replace(None).unwrap();
```

Comment:
Using `Option::take` instead:
```suggestion
        } else if let Some(batch) = self.bound_record_batch.take() {
            self.runtime.block_on(async {
```

Reply:
Is there a reason you didn't include the `if let` suggestion?

Reply:
no, missed that. fixed.